internal/tmpl/action/action.go

```diff
@@ -166,7 +166,11 @@ func (a *Ctrl) execAction(n *core.ActionItem) (int, bool, error) {
 
 	//do one last ready check
 	if !c.ActionReady(n.Typ, n.Param) {
-		a.core.Log.NewEvent("queued action is not ready, should not happen; skipping frame", core.LogSimEvent, -1)
+		if n.Typ == core.ActionBurst && c.CurrentEnergy() < c.MaxEnergy() {
+			a.core.Log.NewEvent("insufficient energy for burst", core.LogActionEvent, c.CharIndex(), "have", c.CurrentEnergy())
```

Comment:
if you want to add this log, this should be in the character template, not in actions.

Reply:
aight u figure it out then gg